pkg/tsdb/cloud-monitoring/kinds/dataquery/types_dataquery_gen.go

```diff
@@ -136,58 +88,45 @@ type AlignmentTypes string
-	AlignmentPeriod *string `json:"alignmentPeriod,omitempty"`
-
-	// Reducer applied across a set of time-series values. Defaults to REDUCE_NONE.
-	CrossSeriesReducer string `json:"crossSeriesReducer"`
-
-	// Array of filters to query data by. Labels that can be filtered on are defined by the metric.
-	Filters []string `json:"filters,omitempty"`
-
-	// Array of labels to group data by.
-	GroupBys []string `json:"groupBys,omitempty"`
-
-	// Alignment function to be used. Defaults to ALIGN_MEAN.
-	PerSeriesAligner *string `json:"perSeriesAligner,omitempty"`
-
-	// Preprocessor is not part of the API, but is used to store the preprocessor and not affect the UI for the rest of parameters
-	Preprocessor *AnnotationQueryPreprocessor `json:"preprocessor,omitempty"`
-
-	// GCP project to execute the query against.
-	ProjectName string `json:"projectName"`
-
-	// Only present if a preprocessor is selected. Alignment period to use when regularizing data. Defaults to cloud-monitoring-auto.
-	SecondaryAlignmentPeriod *string `json:"secondaryAlignmentPeriod,omitempty"`
-
-	// Only present if a preprocessor is selected. Reducer applied across a set of time-series values. Defaults to REDUCE_NONE.
-	SecondaryCrossSeriesReducer *string `json:"secondaryCrossSeriesReducer,omitempty"`
-
-	// Only present if a preprocessor is selected. Array of labels to group data by.
-	SecondaryGroupBys []string `json:"secondaryGroupBys,omitempty"`
-
-	// Only present if a preprocessor is selected. Alignment function to be used. Defaults to ALIGN_MEAN.
-	SecondaryPerSeriesAligner *string `json:"secondaryPerSeriesAligner,omitempty"`
+	// TimeSeriesList Time Series List sub-query properties.
+	TimeSeriesList
 
 	// Annotation text.
 	Text *string `json:"text,omitempty"`
 
 	// Annotation title.
 	Title *string `json:"title,omitempty"`
-
-	// Data view, defaults to FULL.
-	View                 *string                `json:"view,omitempty"`
-	AdditionalProperties map[string]interface{} `json:"-"`
 }
 
-// Preprocessor is not part of the API, but is used to store the preprocessor and not affect the UI for the rest of parameters
-type AnnotationQueryPreprocessor string
-
 // CloudMonitoringQuery defines model for CloudMonitoringQuery.
 type CloudMonitoringQuery struct {
+	// DataQuery These are the common properties available to all queries in all datasources.
+	// Specific implementations will *extend* this interface, adding the required
+	// properties for the given context.
+	DataQuery
+
 	// Aliases can be set to modify the legend labels. e.g. {{metric.label.xxx}}. See docs for more detail.
 	AliasBy *string `json:"aliasBy,omitempty"`
 
+	// Time interval in milliseconds.
+	IntervalMs *float32 `json:"intervalMs,omitempty"`
+
+	// SLO sub-query properties.
+	SloQuery *SLOQuery `json:"sloQuery,omitempty"`
+
+	// GCM query type.
+	// queryType: #QueryType
+	// Time Series List sub-query properties.
+	TimeSeriesList *interface{} `json:"timeSeriesList,omitempty"`
```

Comment:
@spinillos why has this been replaced with an interface rather than with the `CloudMonitoringQueryTimeSeriesList` struct? I'm assuming because the property is defined as a union: `timeSeriesList?: #TimeSeriesList | #AnnotationQuery`?

Reply:
Previously we were expanded the references, so it was creating a struct with the merge of bot models. If you see [here](https://github.com/grafana/grafana/pull/68963/files#diff-1d0d830b738c030297235e3ecfd352c567b5fce71cd4967f5d89eafebdfce556L305-L309): it was merging `#AnnotationQuery` inside  `#TimeSeriesList`.

Without expand the references, by default in go, we are setting an interface. The code was done [here](https://github.com/grafana/thema/blob/main/encoding/gocode/decorators.go#L104). This modifies the default behaviour when we generate the Go models because it wasn't the desired one. Setting interfaces was the less painful thing that we had at this moment but we should improve that to avoid to loose the concrete values.